### PR TITLE
Add API wrapper utility for client

### PR DESCRIPTION
**Pull request by yousefa00**
This PR adds an API wrapper to be used in the client for making http calls and handling responses and errors.

### CODE REVIEW

File: web/api/backend.go [hunk 1]
```diff
@@ -66,6 +71,7 @@ func TokenHandler(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
+	w.WriteHeader(200)
```
Comment (melyxlin) [inline web/api/backend.go]: Not necessary. If it returns something back that is not a error, it will automatically be a StatusOK header

**Review by melyxlin — CHANGES_REQUESTED** [review]
Looks good. Just added minor comments on unnecessary things

File: web/api/backend.go (hunk 1 above)
Comment (yousefa00) [reply web/api/backend.go]: 👍

**Review by melyxlin — APPROVED** [review]

**Review by leorcodes — APPROVED** [review]
LGTM

**Pull request merged**